protocols/imap/src/main/java/org/apache/james/imap/decode/parser/CreateCommandParser.java

```diff
@@ -63,7 +67,16 @@ protected ImapMessage decode(ImapCommand command, ImapRequestLineReader request,
             }
         }
         request.eol();
+        assertMailboxNameJustContainDelimiter(mailboxName, mailboxSession.getPathDelimiter());
         return new CreateRequest(command, mailboxName, tag);
     }
 
+    private void assertMailboxNameJustContainDelimiter(String mailboxName, char delimiter) throws DecodingException {
+        Splitter.on(delimiter)
+            .splitToList(mailboxName)
+            .stream()
+            .filter(s -> !Strings.isNullOrEmpty(s))
```

Comment:
anyMatch

Reply:
I tried but current code is better because I would like to throw exception if all values on list are null or empty